akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst

```diff
@@ -216,6 +216,24 @@ Actor DSL deprecation
 Actor DSL is a rarely used feature and thus will be deprecated and removed.
 Use plain ``system.actorOf`` instead of the DSL to create Actors if you have been using it.
 
+ExtensionKey Deprecation
+------------------------
+
+``ExtensionKey`` is a shortcut for writing :ref:`extending-akka-scala` but extensions created with it
+can not be used from Java and it does in fact not save many lines of code over directly implementing ``ExtensionId``.
+
+
+Old::
+
+  object MyExtension extends ExtensionKey[MyExtension]
+
+New::
+
+  object MyExtension extends ExtensionId[MyExtension] {
+    override def createExtension(system: ExtendedActorSystem): MyExtension =
+      new MyExtension(system)
+  }
```

Comment:
this is not correct, it should `extends ExtensionId[MyExtension] with ExtensionIdProvider` and also implement the `get` and `lookup` methods. It's important to override `get` because otherwise the return type will be wrong from java. See http://doc.akka.io/docs/akka/2.4/scala/extending-akka.html